flink-connector-aws/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumerTest.java

```diff
@@ -312,6 +316,148 @@ public void testListStateChangedAfterSnapshotState() throws Exception {
+        // setup config, initial state and expected state snapshot
+        // ----------------------------------------------------------------------
+        Properties config = TestUtils.getStandardProperties();
+
+        ArrayList<Tuple2<StreamShardMetadata, SequenceNumber>> initialState = new ArrayList<>(1);
+        initialState.add(
+                Tuple2.of(
+                        KinesisDataFetcher.convertToStreamShardMetadata(
+                                new StreamShardHandle(
+                                        "fakeStream1",
+                                        new Shard()
+                                                .withShardId(
+                                                        KinesisShardIdGenerator
+                                                                .generateFromShardOrder(0)))),
+                        new SequenceNumber("11")));
+
+        ArrayList<Tuple2<StreamShardMetadata, SequenceNumber>> expectedStateSnapshot1 =
+                new ArrayList<>(1);
+        expectedStateSnapshot1.add(
+                Tuple2.of(
+                        KinesisDataFetcher.convertToStreamShardMetadata(
+                                new StreamShardHandle(
+                                        "fakeStream1",
+                                        new Shard()
+                                                .withShardId(
+                                                        KinesisShardIdGenerator
+                                                                .generateFromShardOrder(0)))),
+                        new SequenceNumber("12")));
+        ArrayList<Tuple2<StreamShardMetadata, SequenceNumber>> expectedStateSnapshot2 =
+                new ArrayList<>(1);
+        expectedStateSnapshot2.add(
+                Tuple2.of(
+                        KinesisDataFetcher.convertToStreamShardMetadata(
+                                new StreamShardHandle(
+                                        "fakeStream1",
+                                        new Shard()
+                                                .withShardId(
+                                                        KinesisShardIdGenerator
+                                                                .generateFromShardOrder(0)))),
+                        new SequenceNumber("13")));
+
+        // ----------------------------------------------------------------------
+        // mock operator state backend and initial state for initializeState()
+        // ----------------------------------------------------------------------
+
+        TestingListState<Tuple2<StreamShardMetadata, SequenceNumber>> listState =
+                new TestingListState<>();
+        for (Tuple2<StreamShardMetadata, SequenceNumber> state : initialState) {
+            listState.add(state);
+        }
+
+        OperatorStateStore operatorStateStore = mock(OperatorStateStore.class);
```

Comment:
Since Mockito is already used in this calss, and this connector is on the deprecation path I ok to use it. However, the test is very long and it is hard to know what is going on. Can we break out into methods and reduce the complexity of the test case for readability.

Reply:
@vahmed-hamdy existing test util currently does not support `getUnionListState`. As pointed by @dannycranmer, this connector is on deprecation path.

Split test into separate cases and extracted some of the common logic to reduce size.